src/main/java/tahub/contacts/logic/parser/CourseCommandParser.java

```diff
@@ -0,0 +1,56 @@
+package tahub.contacts.logic.parser;
+
+import static tahub.contacts.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static tahub.contacts.logic.parser.CliSyntax.PREFIX_CODE;
+import static tahub.contacts.logic.parser.CliSyntax.PREFIX_NAME;
+import static tahub.contacts.model.course.Course.COURSE_CODE_MESSAGE_CONSTRAINTS;
+
+import java.util.stream.Stream;
+
+import tahub.contacts.logic.commands.CourseCommand;
+import tahub.contacts.logic.parser.exceptions.ParseException;
+import tahub.contacts.model.course.Course;
+
+/**
+ * Parses input arguments and creates a new CourseCommand object
+ */
+public class CourseCommandParser implements Parser<CourseCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the CourseCommand
+     * and returns an CourseCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public CourseCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_CODE, PREFIX_NAME);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_CODE, PREFIX_NAME) || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, CourseCommand.MESSAGE_USAGE));
+        }
+
+        String courseCode = argMultimap.getValue(PREFIX_CODE).get();
+        String courseName = argMultimap.getValue(PREFIX_NAME).get();
+
+        if (!Course.isValidCourseCode(courseCode)) {
+            throw new ParseException(COURSE_CODE_MESSAGE_CONSTRAINTS);
+        }
+
+        if (!Course.isValidCourseName(courseName)) {
+            throw new ParseException(Course.COURSE_NAME_MESSAGE_CONSTRAINTS);
+        }
```

Comment:
`COURSE_CODE_MESSAGE_CONSTRAINTS` and `Course.COURSE_NAME_MESSAGE_CONSTRAINTS`, tho both are part of `Course`

Reply:
the error messages are different!